manual/server-examples/src/docs/dev/index.asciidoc

```diff
@@ -4,6 +4,19 @@ Extending the Neo4j Server
 
 The Neo4j Server can be extended by either plugins or unmanaged extensions.
 
+[WARNING]
+====
+When running your own code alongside Neo4j in the same JVM, there are a few things you should keep in mind:
+
+* Don't create or retain more objects than you strictly need to.
+  Large caches are particularly good at promoting many objects to the old generation, thus increasing the need for expensive full garbage collections.
+* Don't use internal Neo4j APIs.
+  They may change in future releases, and they might not always work as intended in the context of your code.
+* Don't use Java object serialization or reflection if you can avoid it.
+  If you can't, make sure that the `-XX:+TrustFinalNonStaticFields` JVM flag is disabled in `neo4j-wrapper.conf`.
+  This also applies to any runtime dependencies your code may have.
```

Comment:
Could you flip sentences two and three around in the last bullet point (or merge sentence three into the first) to make it clearer that avoiding serialization and reflection also applies to runtime dependencies, rather than unsuring _-XX:+TrustFinalNonStaticFields_ is also not set for runtime dependencies (that's what I parsed when first read it).
E.g.
```
* Don't use Java object serialization or reflection in your code or in any runtime dependency you use if you can avoid it.
  If you can't, make sure that the `-XX:+TrustFinalNonStaticFields` JVM flag is disabled in `neo4j-wrapper.conf`.
```